src/core/shape_inference/include/embedding_shape_infer_utils.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+namespace embedding {
+/**
+ * \brief Return a copy of the source shape with the first dimension replaced by provided dim
+ *
+ * \tparam TShape        Shape type
+ * \tparam TDim          Dimension type
+ *
+ * \param op             Pointer to operator.
+ * \param shape_src      The shape to be copied
```

Comment:
It can be renamed to `emb_table_shape`. this function is specific for embedding operator names don't need to be so generic.

Reply:
Ok, done